rootex/core/resource_loader.cpp

```diff
@@ -169,30 +172,48 @@ VisualModelResourceFile* ResourceLoader::CreateVisualModelResourceFile(const Str
 		return nullptr;
 	}
 
+	const aiScene* scene = s_ModelLoader.ReadFile(path,
+	      aiProcess_CalcTangentSpace      |
+		  aiProcess_Triangulate           | 
+		  aiProcess_JoinIdenticalVertices | 
+		  aiProcess_SortByPType);
+    const aiMesh* mesh = scene->mMeshes[0]; //Taking single mesh
+	aiFace* face;
+
 	// File not found in cache, load it only once
-	PANIC(s_ModelLoader.LoadFile(OS::GetAbsolutePath(path).generic_string()) == false, "Model could not be loaded: " + OS::GetAbsolutePath(path).generic_string());
+	PANIC(scene == false, "Model could not be loaded: " + OS::GetAbsolutePath(path).generic_string());
 
 	VertexData vertex;
 
 	Vector<VertexData> vertices;
-	vertices.reserve(s_ModelLoader.LoadedVertices.size());
+	vertices.reserve(mesh->mNumVertices);
 
-	for (auto& v : s_ModelLoader.LoadedVertices)
+	for (unsigned int v = 0; v < mesh->mNumVertices; v++)
```

Comment:
use range based for loop

Reply:
but here the loop is iterating over three different arrays